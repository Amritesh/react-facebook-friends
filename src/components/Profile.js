import React, { Component, PropTypes } from 'react';
import shallowCompare from 'react-addons-shallow-compare';
import emitter from '../utils/emitter';
import '../style/Profile.css';

type ProfileProp = {
  name: string,
  link: string,
  url: string,
  total_count: number
};

class Profile extends Component {

  props: ProfileProp;

  changeHandler = (e: SyntheticInputEvent): void => emitter.emit('search', e.target.value); // eslint-disable-line no-undef

  shouldComponentUpdate = (nextProps: Object): boolean => shallowCompare(this, nextProps);

  render() {
    const { name, link, url, total_count } = this.props;
    const renderSearchBox = () => {
      return total_count === 0
            ? null
            : <input
              placeholder="Search friends"
              className="searchInput"
              onChange={this.changeHandler}
              autoFocus
            />;
    };

    return (
      <div className="flexContainer">
        <div className="profileStyle">
          <div className="headerStyle">
            <a target="_blank" rel="noopener noreferrer" href={link}>
              <img src={url} alt="" className="imgStyle" />
            </a>
            <h1 className="nameStyle">{name}</h1>
            <h3 className="countStyle">You have {total_count} friends on Facebook</h3>
            {renderSearchBox()}
          </div>
        </div>
      </div>
    );
  }
}

export default Profile;
