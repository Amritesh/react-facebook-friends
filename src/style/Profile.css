.profileStyle {
  position: fixed;
  width: inherit;
  min-height: 100vh;
  background: rgb(58, 87, 149);
  color: #fff;
}

.flexContainer {
  flex: 1 1 0%;
  width: 300px;
}

.headerStyle {
  text-align: center;
  top: auto;
  margin: 3em auto;
  padding: 1em;
}

.imgStyle {
  border: 3px solid white;
  border-radius: 3px;
}

.nameStyle {
  margin: 0;
}

.countStyle {
  margin: 0;
  color: rgb(176, 202, 219);
  font-weight: 300;
}

.searchInput {
  color: black;
  margin-top: 5px;
  box-sizing: border-box;
  padding: .5em;
  width: 75%;
  border-radius: 3px;
  border: 1px solid #aaa;
}

@media (max-width: 1050px) {
  .profileStyle {
    position: static;
    width: auto;
    min-height: initial;
  }

  .flexContainer {
    flex: 1 1 100%;
    width: auto;
  }
}