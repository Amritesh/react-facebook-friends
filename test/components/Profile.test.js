import React from 'react';
import renderer from 'react-test-renderer';
import Profile from '../../src/components/Profile';

test('Profile component should renders correctly', () => {
  const props = {
    link: 'link',
    name: 'test',
    total_count: 1,
    url: 'url'
  };
  const component = renderer.create(<Profile {...props} />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
