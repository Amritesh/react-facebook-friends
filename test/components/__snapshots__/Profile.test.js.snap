// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile component should renders correctly 1`] = `
<div
  className="flexContainer"
>
  <div
    className="profileStyle"
  >
    <div
      className="headerStyle"
    >
      <a
        href="link"
        rel="noopener noreferrer"
        target="_blank"
      >
        <img
          alt=""
          className="imgStyle"
          src="url"
        />
      </a>
      <h1
        className="nameStyle"
      >
        test
      </h1>
      <h3
        className="countStyle"
      >
        You have 
        1
         friends on Facebook
      </h3>
      <input
        className="searchInput"
        onChange={[Function]}
        placeholder="Search friends"
      />
    </div>
  </div>
</div>
`;
